src/events/request/RequestEventHandler.ts

```diff
@@ -78,6 +84,33 @@ export default class RequestEventHandler implements EventHandler<'message'> {
 			}
 		}
 
+		const requestLimit = this.requestLimits.get( origin.channel.id );
+		const internalChannelId = this.internalChannels.get( origin.channel.id );
+		const internalChannel = await DiscordUtil.getChannel( internalChannelId );
+
+		if ( requestLimit && requestLimit >= 0 && internalChannel instanceof TextChannel ) {
+			const internalChannelUserMessages = internalChannel.messages.cache
+				.filter( message => message.embeds.length > 0 && message.embeds[0].author.name == origin.author.tag )
+				.filter( message => new Date().valueOf() - message.embeds[0].timestamp.valueOf() <= 86400000 );
```

Comment:
This line throws an error if `message.embeds[0]` is `undefined`. Which can happen if someone other than the bot writes in the channel. Actually we do have messages in our pending request channels at the very top with no embeds, which would cause the bot to crash as soon as a request is posted.

Reply:
Hm thinking about it actually the line above should prevent that... why doesn't it work?

Reply:
Ah, nvm. It crashes because `timestamp` is null, which is the case because I still have old requests without timestamps in my channel. Not sure if this is worth keeping in mind.

Reply:
Yeah, this needs to be changed. We still have timestamp-less messages in our pending requests channels.

Reply:
Actually nvm I'm wrong